src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -1,71 +1,117 @@
 import useResponsiveLayout from '@hooks/useResponsiveLayout';
 import useThemeStyles from '@hooks/useThemeStyles';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as SearchUtils from '@libs/SearchUtils';
 import Navigation from '@navigation/Navigation';
+import variables from '@styles/variables';
 import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import {useSearchRouterContext} from './SearchRouterContext';
 import SearchRouterInput from './SearchRouterInput';
+import SearchRouterList from './SearchRouterList';
 
 const SEARCH_DEBOUNCE_DELAY = 150;
 
 function SearchRouter() {
     const styles = useThemeStyles();
+    const [betas] = useOnyx(`${ONYXKEYS.BETAS}`);
+    const [recentSearches] = useOnyx(ONYXKEYS.RECENT_SEARCHES);
 
     const {isSmallScreenWidth} = useResponsiveLayout();
     const {isSearchRouterDisplayed, closeSearchRouter} = useSearchRouterContext();
 
+    const [textInputValue, setTextInputValue] = useState('');
     const [userSearchQuery, setUserSearchQuery] = useState<SearchQueryJSON | undefined>(undefined);
+    const contextualReportID = useNavigationState<Record<string, {reportID: string}>, string | undefined>((state) => {
+        return state.routes.at(-1)?.params?.reportID;
+    });
+    const sortedRecentSearches = useMemo(() => {
+        return Object.values(recentSearches ?? {}).sort((a, b) => {
+            const dateA = new Date(a.timestamp);
+            const dateB = new Date(b.timestamp);
+            return dateB.getTime() - dateA.getTime();
+        });
+    }, [recentSearches]);
+
+    const {options, areOptionsInitialized} = useOptionsList({
+        shouldInitialize: true,
+    });
+    const searchOptions = useMemo(() => {
+        if (!areOptionsInitialized) {
+            return {recentReports: [], personalDetails: [], userToInvite: null, currentUserOption: null, categoryOptions: [], tagOptions: [], taxRatesOptions: []};
+        }
+        return OptionsListUtils.getSearchOptions(options, '', betas ?? []);
+    }, [areOptionsInitialized, betas, options]);
+
+    const contextualReportData = contextualReportID ? searchOptions.recentReports?.find((option) => option.reportID === contextualReportID) : undefined;
 
     const clearUserQuery = () => {
+        setTextInputValue('');
         setUserSearchQuery(undefined);
     };
 
-    const onSearchChange = debounce((userQuery: string) => {
-        if (!userQuery) {
-            clearUserQuery();
-            return;
-        }
-
-        const queryJSON = SearchUtils.buildSearchQueryJSON(userQuery);
-
-        if (queryJSON) {
-            // eslint-disable-next-line
-            console.log('parsedQuery', queryJSON);
-
-            setUserSearchQuery(queryJSON);
-        } else {
-            // Handle query parsing error
-        }
-    }, SEARCH_DEBOUNCE_DELAY);
+    const onSearchChange = useCallback(
+        debounce((userQuery: string) => {
```

Comment:
@SzymczakJ 
this construct `useCallback(debounce(...` will not pass our linting. I know because I faced this problem in the past. 
Feel free to read more about this lint rule.

The solution will be either to drop `useCallback` for now, and make sure the function inside is not too computation-heavy, or eslint-ignore this line.
Alternatively check this article: https://kyleshevlin.com/debounce-and-throttle-callbacks-with-react-hooks/

For the record I think this warning is more like:
> eslint cannot figure out what this function argument are, so it shouts a warning

than
> this is bad react code